app/models/graph.rb

```diff
@@ -1,5 +1,13 @@
+require 'timeout'
+
 module Graph
   class << self
+    def query(query_string, variables: {}, context: {}, timeout: nil)
+      Timeout.timeout(timeout) do
```

Comment:
What's the takeway from that? Maybe I'm missing it... 

That code is _opening_ a connection, right? So it times out, but you don't have a dangling connection or any in-progress operations.

But during the GraphQL query, there might some in-progress network operations when you abort (as described in the blog post) but then the connection gets reused by the next HTTP request, and 😿 !

(I've never seen this bug in the wild, but it was the impression I got from that blog post.)